File: sources/go/04_array_sum.go
// task 04 array_sum — expected output: 499999500000
// build: go build -o prog 04_array_sum.go    run: ./prog
// note: plain "go build" only; do not set GOGC=off or any other tuning flags
package main

import "fmt"

func main() {
	arr := make([]int64, 1000000)
	for i := 0; i < 1000000; i++ {
		arr[i] = int64(i)
	}

	var total int64
	for i := 0; i < 1000000; i++ {
		total += arr[i]
	}
	fmt.Println(total)
}

File: sources/julia/04_array_sum.jl
# task 04 array_sum — expected output: 499999500000
# build: julia 04_array_sum.jl    run: julia 04_array_sum.jl

function main()
    n = 1000000
    array = Vector{Int64}(undef, n)
    for i in 1:n
        array[i] = i - 1
    end

    total = Int64(0)
    for i in 1:n
        total += array[i]
    end
    println(total)
end

main()
